java/spark/src/main/java/com/lancedb/lance/spark/read/LanceStatistics.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.lancedb.lance.spark.read;
+
+import com.lancedb.lance.spark.LanceConfig;
+import com.lancedb.lance.spark.internal.LanceDatasetAdapter;
+import com.lancedb.lance.spark.utils.Optional;
+
+import org.apache.spark.sql.connector.read.Statistics;
+import org.apache.spark.sql.types.StructType;
+
+import java.util.OptionalLong;
+
+public class LanceStatistics implements Statistics {
+  private final Optional<Long> rowNumber;
+  private final Optional<StructType> schema;
+
+  public LanceStatistics(LanceConfig config) {
+    this.rowNumber = LanceDatasetAdapter.getDatasetRowCount(config);
+    this.schema = LanceDatasetAdapter.getSchema(config);
+  }
+
+  @Override
+  public OptionalLong sizeInBytes() {
+    // TODO: Support quickly get the bytes on disk for the lance dataset
+    // Now use schema to infer the byte size for simple
```

Comment:
I've added per-field storage stats here: https://github.com/lancedb/lance/pull/3328

I started and then backed away from a whole-dataset number because I ran into some questions.

No need to answer these now but something to think about.

Should a whole-dataset number include indices and manifests?  Or just data?  Or one number for just data and one number for everything?

Should it include files that have been removed from the current version but not yet cleaned up?

Should it include columns that have been dropped (these columns may still have data in old fragments)

Reply:
Hi @westonpace the mostly usage of dataset storage size in spark was to choose use broadcast join or not. So the more precise the data size is, the better the optimizer will work.
On the other hand, the Spark optimizer needs to trade off the precision of plan and the computational performance of each rules.

Reply:
The mostly usage of dataset storage size in spark was to choose use broadcast join or not. So if the accurate data size of lance dataset included the dropped column and deleted row only need O(N) io call (N for the number of fragment), I think we can use this api. If not, use the total size of all the data fragments is also fine for this case.